rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala

```diff
@@ -183,18 +187,14 @@ class BuildBundleSequence(
     //   (returns the pair of (outpath, isSystem) )
 
     val inputs: List[InputFile] = sortedPolicies.flatMap { p =>
-      p.technique.agentConfigs.find( _.agentType == agentType) match {
-        case None      => Nil
-        case Some(cfg) =>
-          val inputs = cfg.templates.collect { case template if(template.included) => InputFile(template.outPath, p.technique.isSystem) } ++
-                       cfg.files.collect { case file if(file.included) => InputFile(file.outPath, p.technique.isSystem) }
-          //must replace RudderUniqueID in the paths
-          p.technique.generationMode match {
-            case TechniqueGenerationMode.MultipleDirectives =>
-              inputs.map(i => i.copy(path = Policy.makeUniqueDest(i.path, p)))
-            case _ =>
-              inputs
-          }
+      val inputs = p.technique.agentConfig.templates.collect { case template if(template.included) => InputFile(template.outPath, p.technique.isSystem) } ++
```

Comment:
why not doing it by agentType as before ?

Reply:
I added a "PolicyTechnique" object which is a specialization of a Technique for one agent (on a node). I now build that oject when creating the node configuration, which is the best place to check if all techniques have an agent target matching the node one. And after that, I avoid all further agentType checking (and I have only one available).

Reply:
That's very true ! Thank you